desc/integrals/bounce_integral.py

```diff
@@ -1,26 +1,818 @@
+        #  valasco is not. FFT single valued part than sum afterword with
+        #  multivalued correction part longer term solution may be adding to
+        #  compute metadata single valued term and multivalued correction factor
+        f = [interp_rfft2(Q, f_i[..., jnp.newaxis, :, :], axes=(-1, -2)) for f_i in f]
+        result = _swap_pl(
+            (integrand(*f, B=B, pitch=1 / pitch_inv[..., jnp.newaxis]) / b_sup_z)
+            .reshape(shape)
+            .dot(self._w)
+            * grad_bijection_from_disc(z1, z2)
+        )
+
+        if check:
+            num_pitch = shape[0]
+            num_rho = shape[-3]
+            shape[0] = num_rho
+            shape[-3] = num_pitch
+            _check_interp(
+                # num_alpha, num_rho, num_pitch, num_well, num_quad
+                (1, *shape),
+                *map(_swap_pl, (zeta, b_sup_z, B)),
+                result,
+                list(map(_swap_pl, f)),
+                plot,
+            )
+        return result
+
+    def plot(self, l, pitch_inv=None, **kwargs):
+        """Plot the field line and bounce points of the given pitch angles.
+
+        Parameters
+        ----------
+        l : int
+            Index into the nodes of the grid supplied to make this object.
+            ``rho=grid.compress(grid.nodes[:,0])[l]``.
+        pitch_inv : jnp.ndarray
+            Shape (num_pitch, ).
+            Optional, 1/λ values whose corresponding bounce points on the field line
+            specified by Clebsch coordinate ρ(l) will be plotted.
+        kwargs
+            Keyword arguments into
+            ``desc/integrals/basis.py::PiecewiseChebyshevSeries.plot1d``.
+
+        Returns
+        -------
+        fig, ax
+            Matplotlib (fig, ax) tuple.
+
+        """
+        B = self._B
+        if B.cheb.ndim > 2:
+            B = PiecewiseChebyshevSeries(B.cheb[l], B.domain)
+        if pitch_inv is not None:
+            errorif(
+                pitch_inv.ndim > 1,
+                msg=f"Got pitch_inv.ndim={pitch_inv.ndim}, but expected 1.",
+            )
+            z1, z2 = B.intersect1d(pitch_inv)
+            kwargs["z1"] = z1
+            kwargs["z2"] = z2
+            kwargs["k"] = pitch_inv
+        kwargs.setdefault("hlabel", r"$\alpha = $" + str(self._alpha) + r", $\zeta$")
+        fig, ax = B.plot1d(B.cheb, **_set_default_plot_kwargs(kwargs))
+        return fig, ax
+
+    def plot_theta(self, l, **kwargs):
+        """Plot θ(α, ζ) on the specified flux surface.
+
+        Notes
+        -----
+        The definition of α in B = ∇ρ × ∇α on an irrational magnetic surface
+        implies the true map θ(α, ζ) is multivalued. In particular, following an
+        irrational field, θ grows to ∞ (always non-monotonically) as ζ → ∞.
+        Therefore, it is impossible to approximate this map using single-valued
+        basis functions defined on a bounded subset of ℝ².
```

Comment:
because countinous function on compact set attains maximum and theta is unbounded